File: algorithms/cpp/1_two_sum/TwoSum.cpp
class TwoSum {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        vector<int> res;
        for (int i = 0; i < nums.size(); i++) {
            if (map.find(target - nums[i]) == map.end()) {
                map[nums[i]] = i;
            } else {
                res.push_back(map[target - nums[i]]);
                res.push_back(i);
                return res;
            }
        }
        return res;
    }
};

File: algorithms/java/fun_two_sum_from_two_arrays/TwoSum.java
class TwoSum {
	public int find(int[] a1, int[] a2, int target) {
		if (a1 == null || a2 == null || a1.length == 0 || a2.length == 0) {
			return 0;
		}
		int i = 0;
		int j = a2.length - 1;
		int count = 0;
		while (i < a1.length && j >= 0) {
			if (a1[i] + a2[j] > target) {
				j--;
			} else {
				i++;
				count += j + 1;
			}
		}
		return count;
	}

	public static void main(String[] args) {
		TwoSum solve = new TwoSum();
		int[] a1 = {1, 3, 6};
		int[] a2 = {2, 4, 5};
		System.out.println(solve.find(a1, a2, 7));
	}
}
